app/mailers/delayed_mailer.rb

```diff
@@ -13,113 +14,101 @@ def initialize(assignment_id, deadline_type, due_at)
     self.due_at = due_at
   end
 
+  def find_assignment
+    Assignment.find(self.assignment_id)
+  end
+
+  def sign_up_topics?
+    sign_up_topics = SignUpTopic.where(['assignment_id = ?', self.assignment_id])
+    (!sign_up_topics.nil? && sign_up_topics.count != 0)
+  end
+
   def perform
-    assignment = Assignment.find(self.assignment_id)
+    assignment = find_assignment
     if !assignment.nil? && !assignment.id.nil?
-      if (self.deadline_type == "metareview")
+      if self.deadline_type == "metareview"
         mail_metareviewers
-        if assignment.team_assignment?
-          teamMails = getTeamMembersMail
-          email_reminder(teamMails, "teammate review")
-        end
+        team_mails = find_team_members_email
+        email_reminder(team_mails, "teammate review") unless team_mails.empty?
       end
 
-      if (self.deadline_type == "review")
+      if self.deadline_type == "review"
         mail_reviewers # to all reviewers
       end
 
-      if (self.deadline_type == "submission")
+      if self.deadline_type == "submission"
         mail_signed_up_users # to all signed up users
       end
 
-      if (self.deadline_type == "drop_topic")
-        # sign_up_topics = SignUpTopic.where( ['assignment_id = ?', self.assignment_id])
-        # if(sign_up_topics != nil && sign_up_topics.count != 0)
-        drop_topics # drop topics from teams that only have one member
-        # reminder to signed_up users of the assignment
-        # end
+      if self.deadline_type == "drop_topic"
```

Comment:
This logic was there in the code from previous group working on the code, and as we have talked with Dr Gehringer he did not mention about changing the logic and it was not in our scope to check those logics.